src/cpp/r/R/Tools.R

```diff
@@ -414,6 +414,13 @@ assign(envir = .rs.Env, ".rs.getVar", function(name)
       unlockBinding(name, packageEnv)
       assign(name, new, packageName)
       lockBinding(name, packageEnv)
+
+      if (namespace && package %in% loadedNamespaces()) {
```

Comment:
FWIW, calling `asNamespace()` will also explicitly load that package's namespace if it's not already loaded. I think we'd want to register our hooks unconditionally (ie, load the package if it's not loaded and register our hook), so I think we can remove the `package %in% loadedNamespaces()` check. Does that sound right?